jmclient/jmclient/configure.py

```diff
@@ -377,33 +377,18 @@ def get_p2pk_vbyte():
 
 def validate_address(addr):
     try:
-        assert len(addr) > 2
-        if addr[:2].lower() in ['bc', 'tb']:
-            # Regtest special case
-            if addr[:4] == 'bcrt':
-                if btc.bech32addr_decode('bcrt', addr)[1]:
-                    return True, 'address validated'
-                return False, 'Invalid bech32 regtest address'
-            #Else, enforce testnet/mainnet per config
-            if get_network() == "testnet":
-                hrpreq = 'tb'
-            else:
-                hrpreq = 'bc'
-            if btc.bech32addr_decode(hrpreq, addr)[1]:
-                return True, 'address validated'
-            return False, 'Invalid bech32 address'
-        #Not bech32; assume b58 from here
-        ver = btc.get_version_byte(addr)
-    except AssertionError:
-        return False, 'Checksum wrong. Typo in address?'
-    except Exception:
-        return False, "Invalid bitcoin address"
-    if ver != get_p2pk_vbyte() and ver != get_p2sh_vbyte():
-        return False, 'Wrong address version. Testnet/mainnet confused?'
-    if len(btc.b58check_to_bin(addr)) != 20:
-        return False, "Address has correct checksum but wrong length."
-    return True, 'address validated'
-
+        # automatically respects the network
+        # as set in btc.select_chain_params(...)
+        x = btc.CCoinAddress(addr)
```

Comment:
Note that one quirk of using `bitcoinlib` and `bitcointx` is that it has 'hex-to-bin' function named `x` that is while convenient and used often, makes using variables named `x` dangerous. If you don't use `bitcointx.core.x` than there's no problem, though. Also such problems are catched with static analysis with mypy.

Reply:
Yes. Initially I was considering using `x()` and `lx()` but then decided that since we have obviously a need for that processing everywhere, including in parts of the codebase that don't have access to `jmbitcoin`, it would be better to put it in base support libraries, and I have now removed references to those fns from your lib. But also better not call a var `x`, so addressed in https://github.com/JoinMarket-Org/joinmarket-clientserver/pull/536/commits/1e2fc3848295a05f4ee685fd435f277c378acadc